frontend/src/components/viewers/MetricsVisualizations.tsx

```diff
@@ -84,23 +89,79 @@ export function MetricsVisualizations({
     { staleTime: Infinity },
   );
 
+  const { isSuccess: isHtmlDownloaded, error: htmlError, data: htmlViewerConfigs } = useQuery<
+    HTMLViewerConfig[],
+    Error
+  >(
+    [
+      'htmlViewerConfig',
+      {
+        artifacts: verifiedHtmlArtifacts.map(linkedArtifact => {
+          return linkedArtifact.artifact.getId();
+        }),
+        state: execution.getLastKnownState(),
+        namespace: namespace,
+      },
+    ],
+    () => getHtmlViewerConfig(verifiedHtmlArtifacts, namespace),
+    { staleTime: Infinity },
+  );
+
+  const {
+    isSuccess: isMarkdownDownloaded,
+    error: markdownError,
+    data: markdownViewerConfigs,
+  } = useQuery<MarkdownViewerConfig[], Error>(
+    [
+      'markdownViewerConfig',
+      {
+        artifacts: verifiedMarkdownArtifacts.map(linkedArtifact => {
+          return linkedArtifact.artifact.getId();
+        }),
+        state: execution.getLastKnownState(),
+        namespace: namespace,
+      },
+    ],
+    () => getMarkdownViewerConfig(verifiedMarkdownArtifacts, namespace),
+    { staleTime: Infinity },
+  );
+
   if (
     verifiedClassificationMetricsArtifacts.length === 0 &&
     verifiedMetricsArtifacts.length === 0 &&
+    verifiedHtmlArtifacts.length === 0 &&
+    verifiedMarkdownArtifacts.length === 0 &&
     !v1VisualizationArtifact
   ) {
     return <Banner message='There is no metrics artifact available in this step.' mode='info' />;
   }
 
   return (
     <>
+      {/* Shows first encountered issue on Banner */}
       {v1ViewerConfigError && (
         <Banner
           message='Error in retrieving v1 metrics information.'
           mode='error'
           additionalInfo={v1ViewerConfigError.message}
         />
       )}
+      {!v1ViewerConfigError && htmlError && (
+        <Banner
+          message='Error in retrieving HTML visualization information.'
+          mode='error'
+          additionalInfo={htmlError.message}
+        />
+      )}
+      {!v1ViewerConfigError && !htmlError && markdownError && (
+        <Banner
+          message='Error in retrieving Markdown visualization information.'
+          mode='error'
+          additionalInfo={markdownError.message}
+        />
+      )}
```

Comment:
nit: these start to get more complex, would it be better organizing these in a code struct, e.g. you can do this:

```ts
<>
{(() => {
  if (v1ViewerConfigError) {
    return <Banner .../>
  }
  if (htmlError) {
    return <Banner .../>
  }
  if (markdownError) {
    return <Banner .../>
  }
  return null
})()}
</>
```

the technique is usually called immediately invoked anonymous function, it's useful for embedding more complex code in JSX template.

Reply:
Awesome! Thank you so much for introducing this anonymous function! It is very helpful and good readability.